source/agora/consensus/Fee.d

```diff
@@ -188,16 +192,18 @@ public class FeeManager
 
     /***************************************************************************
 
-        Accumulated fee per public key
+        Accumulated fee per public key and payout height
 
         Note that while we usually use UTXO hashes to refer to validators
         instead of public keys (as two validators could be using the same public
         key, although that is not recommended), the public key is what is used
         to generated the UTXO, hence why it is used as index.
+        The `payout height` is the height of the block with the `Coinbase`
+        transaction.
 
     ***************************************************************************/
 
-    private Amount[PublicKey] accumulated_fees;
+    private Amount[PublicKey][Height] accumulated_fees;
```

Comment:
Don't think we ever clear this, do we ? Having a by height AA at least makes it easier to clean now.

Reply:
The paid height is removed from the AA in `clearAccumulatedFees` 
```D
this.accumulated_fees.remove(height);
```